src/events/base.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Base utilities exposing common functionalities for all Events classes."""
+
+from functools import wraps
+from typing import Callable
+
+from ops import CharmBase, EventBase, Object, StatusBase
+
+from core.context import Context
+from core.domain import SparkServiceAccountInfo, DeployMode, Flavour
+from common.logging import WithLogging
+from core.workload import KafkaAppWorkloadBase
+
+from managers.k8s import K8sManager
+from events import Status
+
+class BaseEventHandler(Object, WithLogging):
+    """Base class for all Event Handler classes in the Spark Integration Hub."""
+
+    workload: KafkaAppWorkloadBase
+    charm: CharmBase
+    context: Context
+
+    def get_app_status(
+        self,
+        service_account: SparkServiceAccountInfo | None,
+    ) -> StatusBase:
+        """Return the status of the charm."""
+        if not self.workload.ready():
+            return Status.WAITING_PEBBLE.value
+
+        if not service_account:
+            return Status.MISSING_INTEGRATION_HUB.value
+
+        k8s_manager = K8sManager(service_account_info=service_account, workload=self.workload)
+
+        if not k8s_manager.has_cluster_permissions():
+            return Status.INSUFFICIENT_CLUSTER_PERMISSIONS.value
+
+        # If the flavour Kafka is enabled, Kafka relation needs to exists
+        if (
+            self.context.config.flavour == Flavour.KAFKA.value and
+            not self.context.kafka
+        ):
+            return Status.MISSING_KAFKA.value
+
+        # If metastore is enabled, so need to be the object storage
+        if (
+            self.context.metastore and
```

Comment:
I did not understand how the existence of object storage relation is mandated due to metastore being enabled. Is this branch supposed to be equivalent to `self.context.config.flavour != Flavour.KAFKA.value`?

Reply:
when we enable the metastore, it means that we want to consolidate the results into a Hive Table and for storing the raw date we need to have access to an object storage backend (s3 or azure). 

I agree, it is somewhat obscure with the current comment, so I have explained this better in the comment to improve the clarity

Reply:
But even if we don't have PostgreSQL as metastore, we are still storing the raw data in the object storage backend, no? In that case, shouldn't the object storage be available at all times?